File: boilerplate/java/order-service/src/main/java/com/example/orderservice/infrastructure/persistence/OrderEntity.java
package com.example.orderservice.infrastructure.persistence;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.OffsetDateTime;
import java.util.List;
import java.util.UUID;

import com.example.orderservice.domain.models.OrderState;

@Entity
@Table(name = "orders")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class OrderEntity {
    @Id
    private UUID id;
    private UUID customerId;
    private OffsetDateTime createdAt;
    private OffsetDateTime confirmedAt;
    private OffsetDateTime deletedAt;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private OrderState status;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderItemEntity> items;
}

File: boilerplate/nestjs/order-service/src/infrastructure/persistence/order.entity.ts
// infrastructure/persistence/order.entity.ts
import {
  Entity,
  Column,
  PrimaryColumn,
  CreateDateColumn,
  OneToMany,
} from "typeorm";

import { OrderItemEntity } from "./order-item.entity";

@Entity("orders")
export class OrderEntity {
  @PrimaryColumn("uuid")
  id: string;

  @Column({ name: "customer_id", type: "uuid", nullable: true })
  customerId: string;

  @Column({ name: "status", type: "varchar", nullable: false })
  status: string;

  @CreateDateColumn({ name: "created_at" })
  createdAt: Date;

  @Column({ name: "confirmed_at", type: "timestamptz", nullable: true })
  confirmedAt: Date | null;

  @Column({ name: "deleted_at", type: "timestamptz", nullable: true })
  deletedAt: Date | null;

  @OneToMany(() => OrderItemEntity, (item) => item.order, {
    cascade: true,
    orphanedRowAction: "delete",
  })
  items: OrderItemEntity[];
}
